docs/pages/docs/routing/middleware.mdx

```diff
@@ -343,6 +343,42 @@ export default createMiddleware({
   APIs](/docs/routing/navigation#localized-pathnames) in your components.
 </Callout>
 
+<details>
+<summary>How to revalidate localized pathnames?</summary>
+
+Depending on how the route is generated, whether statically or dynamically, if you're attempting to use Next.js's function `revalidatePath()`, you will need to call it with two different pathnames.
+
+Let's take an example with this folder structure:
+```
+.
+└── [locale]/
+    └── news/
+        └── [slug]
+```
+
+And, this localized configuration :
+```tsx filename="middleware.ts"
+import createMiddleware from 'next-intl/middleware';
+ 
+export default createMiddleware({
+  defaultLocale: 'en',
+  locales: ['en', 'fr'],
+  pathnames: {
+    '/news/[slug]': {
+      en: '/news/[slug]',
+      fr: '/infos/[slug]'
+    },
+  }
+});
+```
+
+Assuming the pathname is `/fr/infos/some-article`, if the page is dynamically generated, you will need to revalidate like this: `revalidatePath('/fr/infos/some-article')`. However, if the page is statically rendered, the path to revalidate is this one: `revalidatePath('/fr/news/some-article')`. So, if you don't know if the page is statically or dynamically generated, we advise you to revalidate both.
+
+You can find more information on the issue [here](https://github.com/amannn/next-intl/issues/846).
```

Comment:
Maybe we can link to the Next.js issue? I'd like to close the issue on the `next-intl` side once your PR here is merged.

```suggestion
See also [`vercel/next.js#59825`](https://github.com/vercel/next.js/issues/59825).
```